CMakeLists.txt

```diff
@@ -0,0 +1,15 @@
+cmake_minimum_required(VERSION 3.12)
+project(ping-cpp)
+
+set(CMAKE_CXX_FLAGS "-std=c++11 -Wall -Werror")
+include_directories(src/device src/message)
+
+set(DEVICE_SRC
+  src/device/ping-device.cpp
+  src/device/ping-port-linux.cpp
+  src/device/ping-time-linux.cpp
```

Comment:
Does not make sense to have such things here, this library should be OS independent.

Reply:
> this library should be OS independent.

We do not have any requirement to support Windows Mac or anything else here. It is just code that works.

Beside, I designed this code to be used in arduino projects (OS independent). We can have ping-time-arduino.cpp and ping-port-arduino.cpp, they are very easy to implement.

Reply:
> there is libraries that handle it that exist before we are born, such libraries are good and compatible with all OSs.

@patrickelectric I should have also mentioned: I intend to do a ping-port-qt like you mention here. I just had not used QSerial + cmake when I made this, so I used the linux port implementation I already had as a 'stub' in order to test the device class (which this pr is about). I will make the arduino and qt implementations of ping-serial and ping-qt next.

Reply:
We should avoid Qt here IMO, Qt is a good toolkit/library for development of applications but not as good for libraries, we should aim to use boost since it's a std candidate and follows std guidelines.

Reply:
We cannot use std::booost serial port on arduino or stm32.

Reply:
ok that's fine.